Packs/DatadogCloudSIEM/Integrations/DatadogCloudSIEM/DatadogCloudSIEM.py

```diff
@@ -0,0 +1,791 @@
+            api_instance.list_events(
+                start=int(start_time.timestamp() if start_time else 0),
+                end=int(end_time.timestamp() if end_time else 0),
+            )
+        except Exception:
+            return "Authorization Error: Invalid Application Key."
+        return "ok"
+
+
+def create_event_command(
+    configuration: Configuration, args: Dict[str, Any]
+) -> Union[CommandResults, DemistoException]:
+    """
+    Creates an event in Datadog.
+
+    Args:
+        configuration (Configuration): The configuration object for Datadog.
+        args (Dict[str, Any]): A dictionary of arguments for creating the event.
+
+    Returns:
+        CommandResults: A CommandResults object with the following properties:
+        - "readable_output": A human-readable message indicating whether the event was created successfully.
+        - "Event": A dictionary representing the created event.
+    """
+    priority = args.get("priority")
+    alert_type = args.get("alert_type")
+    if priority and priority not in EventPriority.allowed_values:
+        return DemistoException("Priority not in allowed values.")
+    if alert_type and alert_type not in EventAlertType.allowed_values:
+        return DemistoException("Alert type not in allowed values.")
+    date_happened = args.get("date_happened")
+
+    if date_happened:
+        date_happened_timestamp = parse(date_happened, settings={"TIMEZONE": "UTC"})
+        if not is_within_time(
+            int(date_happened_timestamp.timestamp() if date_happened_timestamp else 0)
+        ):
+            return CommandResults(
+                readable_output="The time of the event cannot be older than 18 hours!\n"
+            )
+    date_happened = (
+        parse(date_happened, settings={"TIMEZONE": "UTC"}) if date_happened else None
+    )
+    event_body = {
+        "title": args.get("title"),
+        "text": args.get("text"),
+        "tags": args.get("tags", []).split(",") if args.get("tags") else None,
+        "alert_type": EventAlertType(args.get("alert_type")),
+        "priority": EventPriority(args.get("priority")),
+        "aggregation_key": args.get("aggregation_key"),
+        "related_event_id": int(args.get("related_event_id", 0))
+        if args.get("related_event_id")
+        else None,
+        "host": args.get("host_name"),
+        "device_name": args.get("device_name"),
+        "date_happened": int(date_happened.timestamp()) if date_happened else None,
+        "source_type_name": args.get("source_type_name"),
+    }
+    body = EventCreateRequest(
+        **{key: value for key, value in event_body.items() if value is not None}
+    )
+
+    with ApiClient(configuration) as api_client:
+        api_instance = EventsApi(api_client)
+        response = api_instance.create_event(body=body)
+        readable_output = (
+            "Event created successfully!"
+            if response and response.status == "ok"
+            else ERROR_MSG
```

Comment:
Can you give a more elaborate error than something went wrong ? can we get what went wrong from the API ?